src/main/java/cz/xtf/openshift/OpenShiftUtil.java

```diff
@@ -828,4 +786,126 @@ private Path storeLog(String log, Path dirPath, String fileName) throws IOExcept
 
 		return filePath;
 	}
+
+	private OpenShiftWaiters waiters = new OpenShiftWaiters();
+
+	// Waiting
+	public OpenShiftWaiters waiters() {
+		return waiters;
+	}
+
+	public class OpenShiftWaiters {
+
+		private OpenShiftWaiters() {
+
+		}
+
+		/**
+		 * Creates waiter for build completion with preconfigured timeout 10 minutes,
+		 * 5 seconds interval check and both logging points.
+		 *
+		 * @param buildName name of build to be waited upon.
+		 * @return Waiter instance
+		 */
+		public Waiter buildCompletion(String buildName) {
+			String reason = "Waiting for completion of build " + buildName;
+			Supplier<String> supplier =  () -> getBuild(buildName).getStatus().getPhase();
+
+			return new SupplierWaiter<>(supplier, "Complete"::equals, "Failed"::equals, TimeUnit.MINUTES, 10, reason).logPoint(Waiter.LogPoint.BOTH).interval(5_000);
+		}
+
+		/**
+		 * Creates waiter for clean project with 20 seconds timeout.
+		 *
+		 * @return Waiter instance
+		 */
+		public Waiter cleanProject() {
+			BooleanSupplier bs = () -> {
+				List<Boolean> cleanedResources = new ArrayList<>();
+				cleanedResources.add(getDeploymentConfigs().isEmpty());
+				cleanedResources.add(client.replicationControllers().list().getItems().isEmpty());
+				cleanedResources.add(getBuildConfigs().isEmpty());
+				cleanedResources.add(getImageStreams().isEmpty());
+				cleanedResources.add(getEndpoints().isEmpty());
+				cleanedResources.add(getServices().isEmpty());
+				cleanedResources.add(getBuilds().isEmpty());
+				cleanedResources.add(getRoutes().isEmpty());
+				cleanedResources.add(getPods().isEmpty());
+				cleanedResources.add(getPersistentVolumeClaims().isEmpty());
+				cleanedResources.add(getHorizontalPodAutoscalers().isEmpty());
+				cleanedResources.add(getConfigMaps().isEmpty());
+				cleanedResources.add(getUserSecrets().isEmpty());
+				cleanedResources.add(getUserServiceAccounts().isEmpty());
+
+				return !cleanedResources.contains(false);
+			};
+
+			return new SimpleWaiter(bs, TimeUnit.SECONDS,20,"Cleaning project.");
+		}
+
+		/**
+		 * Creates a waiter that checks whether all deployment pods are ready.
+		 * Defaults to 2 minutes timeout.
+		 *
+		 * @param dcName name of deploymentConfig
+		 * @return Waiter instance
+		 */
+		public Waiter deploymentReadiness(String dcName) {
```

Comment:
Thinking whether this naming is fluent enough. Typical usage then would be: 

openshift.waiters().deploymentReadiness(args...).run();

Suggestions?

Reply:
Maybe something more descriptive, e.g.

```
openshift.waiters().isReady().dc(arg).run();
// where isReady() defines condition
// dc() is self-explaining :)
```
or
```
openshift.waiters().isReady().withResource(arg).run();
````
or 
```
openshift.waiters().isReady().resource(arg).run();
```

Reply:
Renamed `Waiter` creating methods so they start with either with is, has or does(eg. `doesUrlReturnsOK`). Ability to specify request with like `isReady()` could be good once we have more then few default waiters.